crates/polars-io/src/options.rs

```diff
@@ -16,15 +16,18 @@ pub struct RowIndex {
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct HiveOptions {
-    pub enabled: bool,
+    /// This can be `None` to automatically enable for single directory scans
+    /// and disable otherwise. However it should be initialized if it is inside
+    /// a DSL / IR plan.
+    pub enabled: Option<bool>,
```

Comment:
Maybe this should become an Enum like we did for `JoinCoalesce`? Not sure about best practices here.

Reply:
Could be both. Not super important to me (this saves somes serde bloat), the other is more readable in matching.

In any case. I am going to merge this so we can work towards the RC. :)